nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/main/java/org/apache/nifi/processor/util/bin/Bin.java

```diff
@@ -173,4 +174,19 @@ public long getBinAge() {
         final long ageInNanos = System.nanoTime() - creationMomentEpochNs;
         return TimeUnit.MILLISECONDS.convert(ageInNanos, TimeUnit.NANOSECONDS);
     }
+
+    /**
+     * @return the UUID of the flow file generated by the merge of this bin's content
+     */
+    public String getUuid() {
+        return uuid;
+    }
+
+    /**
+     * Set the UUID of the flow file generated by the merge of this bin's content
+     * @param uuid UUID to set
+     */
+    public void setUuid(String uuid) {
```

Comment:
I just had another look and I'm not sure to see a way to do it nicely. If you have an idea, let me know, I'll be happy to give it a try.